src/lib/fields/VocabularyField.js

```diff
@@ -65,20 +65,23 @@ export class VocabularyField extends Field {
 }
 
 export class AllowAdditionsVocabularyField extends VocabularyField {
-  deserialize(record) {
-    const fieldValue = _get(record, this.fieldpath, this.deserializedDefault);
-    // We deserialize the values in the format
-    // {id: 'vocab_id', <labelField>: 'vacab_name'} for controlled values
-    // and {<labelField>: 'vocab_name'} for user added entries
-    const _deserialize = (value) => ({
+  // We deserialize the values in the format
+  // {id: 'vocab_id', <labelField>: 'vocab_name'} for controlled values
+  // and {<labelField>: 'vocab_name'} for user added entries
+  _deserialize(value) {
+    return {
       ...(value.id ? { id: value.id } : {}),
       [this.labelField]: value[this.labelField],
-    });
+    };
+  }
+
+  deserialize(record) {
+    const fieldValue = _get(record, this.fieldpath, this.deserializedDefault);
     let deserializedValue = null;
     if (fieldValue !== null) {
       deserializedValue = Array.isArray(fieldValue)
-        ? fieldValue.map(_deserialize)
-        : _deserialize(fieldValue);
+        ? fieldValue.map((value) => this._deserialize(value, this))
+        : this._deserialize(fieldValue, this);
```

Comment:
We don't need to pass `this` as an argument, right?

Reply:
Hmm, I passed it to have access to `this.labelField` (line 74), otherwise we do not have access to it, or do we?

Reply:
We do now that `_deserialize` is a method :) . No need to specify it in the method (more like Java as opposed to Python) For more: https://javascript.info/object-methods#this-in-methods

Reply:
I see, thanks!